components/site-search-trigger.js

```diff
@@ -0,0 +1,250 @@
+// node_modules
+import { MagnifyingGlassIcon, XCircleIcon } from "@heroicons/react/20/solid";
+import { Dialog } from "@headlessui/react";
+import { useRouter } from "next/router";
+import PropTypes from "prop-types";
+import { useEffect, useRef, useState } from "react";
+// components
+import { useSessionStorage } from "./browser-storage";
+// lib
+import { KC, UC } from "../lib/constants";
+import { encodeUriElement } from "../lib/query-encoding";
+
+// Maximum number of recent terms to keep track of.
+const MAX_RECENT_TERMS = 5;
+
+/**
+ * Hook to process the array of recently searched terms stored in sessionStorage. The most recent
+ * term gets stored first in the array.
+ */
+function useRecentSearches() {
+  const [recentTermsJson, setRecentTermsJson] = useSessionStorage(
+    "recent-site-search",
+    JSON.stringify([])
+  );
+  const recentTerms = JSON.parse(recentTermsJson);
+
+  function saveRecentTerm(recentTerm) {
+    // If the new search term is already in the list, remove it so we can then move it to the
+    // front of the list.
+    const updatedRecentTerms = recentTerms.includes(recentTerm)
+      ? recentTerms.filter((term) => term !== recentTerm)
+      : recentTerms;
+
+    // Add the new term to the front of the list, and limit the length of the list -- older items
+    // fall off the bottom of the list.
+    updatedRecentTerms.unshift(recentTerm);
+    const limitedRecentTerms = JSON.stringify(
+      updatedRecentTerms.slice(0, MAX_RECENT_TERMS)
+    );
```

Comment:
Should there be a way to clear this list manually? Maybe not worth worrying about for this particular PR, but was something I thought of

Reply:
I don’t know actually. It crossed my mind to put an X next to each to clear individual items, but I couldn’t explain to myself why that would be useful.